src/parsing_routines_matrixnetworks.jl

```diff
@@ -1,123 +1,123 @@
 """
-    For loading networks with stoichiometry stored in matrices.
-    Assumed that the substrate and product stoichiometry matrices
-    are stored as numspecies by numrxs matrices, with entry (i,j)
-    giving the stoichiometric coefficient of species i within rx j.
+Given substrate and product stoichiometric matrices ,rate-expressions and list of species,parameters
+, return a ReactionSystem that describes the chemical reaction network
+
+Assumed that the substrate and product stoichiometry matrices
+are stored as numspecies by numrxs matrices, with entry (i,j)
+giving the stoichiometric coefficient of species i within rx j.
 """
 
+struct MatrixNetwork{S ,T ,U ,V ,W,X}
+    """The symbolic expressions for each reaction rate."""
+    rateexprs::S
+
+    """substoich[i,j] = is the substrate stoichiometric coefficient matrix"""
+    substoich::T
 
-""" 
-    For dense matrices
-"""
-function loadrxnetwork(::MatrixNetwork, 
-                        rateexprs::AbstractVector, 
-                        substoich::AbstractMatrix, 
-                        prodstoich::AbstractMatrix; 
-                        species::AbstractVector=Any[], 
-                        params::AbstractVector=Any[])
-
-    sz = size(substoich)
-    @assert sz == size(prodstoich)
-    numspecs = sz[1]
-    numrxs = sz[2]
+    """prodstoich[i,j] is the product stoichiometric coefficient matrix"""
+    prodstoich::U
+
+    """species in the network """
+    species::V
+
+    """ Parameters """
+    params::W
+
+    """independent variable, time """
+    t::X
+end
+MatrixNetwork(rateexprs,substoich,prodstoich; species=Any[],params=Any[],t=nothing) =
+           MatrixNetwork(rateexprs,substoich,prodstoich,species,params,t)
+
+# for dense matrices
+function loadrxnetwork(mn::MatrixNetwork{S,T,U,V,W,X}) where {S <: AbstractVector,
+        T <: Matrix, U <: Matrix{Int}, V <: AbstractVector, W <: AbstractVector,X <: Any}
+
+    sz = size(mn.substoich)
+    @assert sz == size(mn.prodstoich)
+     numrxs= sz[1]
+    numspecs = sz[2]
 
     # create the network
-    rn = make_empty_network()        
-    t  = ModelingToolkit.get_iv(rn)
+    rn = make_empty_network()
+    t = (mn.t === nothing) ? (@variables t)[1] : mn.t
 
-    # create the species if none passed in        
-    isempty(species) && (species = [funcsym(:S,t,i) for i=1:numspecs])
+    # create the species if none passed in
+    species = isempty(mn.species) ? [funcsym(:S,t,i) for i = 1:numspecs] : mn.species
     foreach(s -> addspecies!(rn, s, disablechecks=true), species)
 
     # create the parameters
-    foreach(p -> addparam!(rn, p, disablechecks=true), params)
+    foreach(p -> addparam!(rn, p, disablechecks=true), mn.params)
 
     # create the reactions
     # we need to create new vectors each time as the ReactionSystem
     # takes ownership of them
-    for j = 1:numrxs
+    for i = 1:numrxs
```

Comment:
Why did you flip `i` and `j`? Shouldn't we keep columns as reactions? Is this to be consistent with Catalyst? I guess the problem is actually Catalyst which should use num species by num reactions but was defined transposed. I think we should just fix the definition there and then make a breaking release.